### PR TITLE
Story/11994

### PR DESCRIPTION
**Pull request by williamsajr**

### CODE REVIEW

**Review by apatrick88 — APPROVED** [review]

**Review by davidmcgaw — APPROVED** [review]

File: addons/udes_stock/models/stock_picking_batch.py [hunk 1]
```diff
@@ -283,10 +283,28 @@ def get_next_task(
...
+        """Get the next not completed task of the batch to be done.
+        Expect a singleton.
         """
-        Get the next not completed task of the batch to be done.
+        task = self.get_next_tasks(
+            skipped_product_ids=skipped_product_ids,
+            skipped_move_line_ids=skipped_move_line_ids,
+            task_grouping_criteria=task_grouping_criteria,
+            limit=1
+        )[0]
```
Comment (ws-mps) [inline addons/udes_stock/models/stock_picking_batch.py:296]: What happens if there isn't any next task, won't `[]` break?

Comment (ws-mps) [reply addons/udes_stock/models/stock_picking_batch.py:296]: Saw the empty task thing.

File: addons/udes_stock/controllers/stock_picking_batch.py [hunk 2]
```diff
@@ -118,6 +118,76 @@ def get_next_task(
...
+        next task: picking_id and package/product information.
+        Returns an empty list otherwise.
+        """
+        batch = _get_batch(request.env, ident)
+
+        with batch.statistics() as stats:
+            tasks = batch.get_next_tasks(
+                skipped_product_ids=skipped_product_ids,
+                skipped_move_line_ids=skipped_move_line_ids,
+                limit=limit
```
[the remark this answers is not in the document]
Comment (williamsajr) [reply addons/udes_stock/controllers/stock_picking_batch.py:150]: Yeah, I wasn't sure how to handle this. On one hand I don't think any mobile user would want to see 100+ tasks at once but on the other hand it's an annoying limitation. 
I could change this to have a default limit of False (meaning unlimited) and change some of the code on the model.

[the remark this answers is not in the document]
Comment (williamsajr) [reply addons/udes_stock/controllers/stock_picking_batch.py:150]: ^ Added this as a separate commit

**Review by kenvac — APPROVED** [review]

**Pull request merged**